pydantic/json_schema.py

```diff
@@ -741,7 +740,7 @@ def default_schema(self, schema: core_schema.WithDefaultSchema) -> JsonSchemaVal
             default = schema['default']
         elif 'default_factory' in schema:
             default = schema['default_factory']()
-        else:
+        else:  # pragma: no cover
```

Comment:
Should we use `assert_never` instead of the `ValueError`? Can this be reached?

Reply:
It could be also done like this
```py
if condition1:
elif condition2:
else:
    assert condition3
```
This gets the code to the full coverage if we have tests for all cases

Reply:
```suggestion
        else:
```